librosa/core/audio.py

```diff
@@ -194,6 +194,10 @@ def load(
     else:
         sr = sr_native
 
+    # SR Cast
+    if isinstance(sr, float) and sr.is_integer():
+        sr = int(sr)
```

Comment:
I don't think this part should be here.  Imagine the case where a user explicitly calls with an integer-valued float target sr:
```
y, sr = librosa.load(somefile, sr=8000.0)
```
With this logic in place, the output sr will be integer typed, which is not what the user specified.

I think it's best to leave it as it was.  If the underlying decoder produces an int type, that's fine to annotate, but we shouldn't coerce it here.

Reply:
It makes sense, I will remove this part.